### PR TITLE
base: Remove dockerd from the base image

### LINKED ISSUE #436: [Image Update] Remove docker or add support for root-less docker
#### Which image would you like to change?

Base

#### What changes are you requesting?

Either remove the dockerd and related packages or add support for running docker in rootless mode: https://docs.docker.com/engine/security/rootless/#daemon

#### What scenario are these changes enabling?

By removing we reduce the image size.

### PR DESCRIPTION
**Pull request by surajssd**
A user cannot start the daemons like dockerd or related services since they don't have permissions to do so.

Fixes #436

### CODE REVIEW

File: tests/command_list [hunk 1]
```diff
@@ -257,10 +251,6 @@ dltest
 dmesg
 dnsdomainname
 do
-docker
```
[the remark this answers is not in the document]
Comment (priyaananthasankar) [reply tests/command_list:260]: I agree, but due to not having root access, is this CLI totally useless? if so, can remove, else can keep it as it doesn't directly contribute to vulnerabilities anyway

**Review by priyaananthasankar — COMMENTED** [review]
provided comments

File: tests/command_list (hunk 1 above)
[the remark this answers is not in the document]
Comment (surajssd) [reply tests/command_list:260]: `docker login azure` has been deprecated from the latest version of docker cli. All that code was part of https://github.com/docker-archive/compose-cli and has been removed.

As of this writing the docker-cli that is shipped in CloudShell does not have that sub-command:

```bash
$ docker --version
Docker version 24.0.9, build 293681613032e6d1a39cc88115847d3984195c24

$ docker login --help

Usage:  docker login [OPTIONS] [SERVER]

Log in to a registry.
If no server is specified, the default is defined by the daemon.

Options:
  -p, --password string   Password
      --password-stdin    Take the password from stdin
  -u, --username string   Username
```

[the remark this answers is not in the document]
Comment (jepio) [reply tests/command_list:260]: alright, makes sense

**Review by jepio — APPROVED** [review]

**Review by priyaananthasankar — APPROVED** [review]

**Pull request merged**